src/pip/__pip-runner__.py

```diff
@@ -30,7 +32,21 @@ def find_spec(
         return spec
 
 
+def check_python_version() -> None:
+    # Import here to ensure the imports happen after the sys.meta_path change.
+    from pip._vendor.packaging.specifiers import SpecifierSet
+    from pip._vendor.packaging.version import Version
+
+    py_ver = Version("{0.major}.{0.minor}.{0.micro}".format(sys.version_info))
+    if py_ver not in SpecifierSet(PYTHON_REQUIRES):
+        raise SystemExit(
+            f"This version of pip does not support python {py_ver} "
+            f"(requires {PYTHON_REQUIRES})"
+        )
```

Comment:
```suggestion
def check_python_version() -> None:
    if sys.version_info < (3, 7):
        raise SystemExit(
            f"This version of pip requires Python 3.7+. You have:\n"
            f"{sys.version}"
        )
```

Reply:
We aren't going to get more complicated anytime in the future, so I wouldn't bother with trying to be fully flexible here TBH.